Sources/PackageRegistry/ChecksumTOFU.swift

```diff
@@ -130,9 +134,69 @@ struct PackageVersionChecksumTOFU {
         }
     }
 
+    // MARK: - manifests
+
+    func validateManifest(
+        registry: Registry,
+        package: PackageIdentity.RegistryIdentity,
+        version: Version,
+        toolsVersion: ToolsVersion?,
+        checksum: String,
+        timeout: DispatchTimeInterval?,
+        observabilityScope: ObservabilityScope,
+        callbackQueue: DispatchQueue,
+        completion: @escaping (Result<Void, Error>) -> Void
+    ) {
+        let contentType = Fingerprint.ContentType.manifest(toolsVersion)
+
+        self.readFromStorage(
```

Comment:
Unlike source archives, registry doesn't provide expected checksum for manifests, so here we do true TOFU I suppose, and that is if checksum not found in storage we save it (whereas for source archive we have registry as an other source).